soc/arm/st_stm32/stm32wba/power.c

```diff
@@ -60,35 +77,80 @@ void set_mode_stop(uint8_t substate_id)
 	}
 }
 
+#if defined(CONFIG_PM_S2RAM)
+static int suspend_to_ram(void)
+{
+	LL_LPM_EnableDeepSleep();
+
+	while (LL_PWR_IsActiveFlag_ACTVOS() == 0) {
+	}
+
+	/* Select mode entry : WFE or WFI and enter the CPU selected mode */
+	k_cpu_idle();
+
+	return 0;
+}
+
+void set_mode_suspend_to_ram(uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	/* Enable SRAM full retention */
+	LL_PWR_SetSRAM1SBRetention(LL_PWR_SRAM1_SB_FULL_RETENTION);
+	LL_PWR_SetSRAM2SBRetention(LL_PWR_SRAM2_SB_FULL_RETENTION);
+
+	/* Enable RTC wakeup
+	 * This configures an internal pin that generates an event to wakeup the system
+	 */
+	LL_PWR_EnableWakeUpPin(LL_PWR_WAKEUP_PIN7);
+	LL_PWR_SetWakeUpPinSignal3Selection(LL_PWR_WAKEUP_PIN7);
+
+	/* Clear flags */
+	LL_PWR_ClearFlag_SB();
+	LL_PWR_ClearFlag_WU();
+	LL_RCC_ClearResetFlags();
+
+	disable_cache();
+
+	/* Select standby mode */
+	LL_PWR_SetPowerMode(LL_PWR_MODE_STANDBY);
+
+	/* Save context and enter Standby mode */
+	arch_pm_s2ram_suspend(suspend_to_ram);
+
+	/* Execution is restored at this point after wake up */
+	/* Restore system clock as soon as we exit standby mode */
+	sys_clock_idle_exit();
+}
+#endif
+
 /* Invoke Low Power/System Off specific Tasks */
 void pm_state_set(enum pm_state state, uint8_t substate_id)
 {
 	switch (state) {
 	case PM_STATE_SUSPEND_TO_IDLE:
 		set_mode_stop(substate_id);
+
+		/* Select mode entry : WFE or WFI and enter the CPU selected mode */
+		k_cpu_idle();
+
+		break;
+	case PM_STATE_SUSPEND_TO_RAM:
+#if defined(CONFIG_PM_S2RAM)
+		set_mode_suspend_to_ram(substate_id);
+#else
+		LOG_DBG("Suspend to RAM needs CONFIG_PM_S2RAM to be enabled");
```

Comment:
if you just want to have a debug message here and not make it fatal, wouldn't be better calling `k_cpu_idle()` ? otherwise will have this looping in the idle thread until this state not be suitable.

Reply:
Put the define before the "case" to fall into default, as it is done for other cases, and for other STM32 series.